exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -587,16 +587,12 @@ func (p *Poloniex) ModifyOrder(ctx context.Context, action *order.Modify) (*orde
 		return nil, err
 	}
 
-	return &order.Modify{
-		Exchange:          action.Exchange,
-		AssetType:         action.AssetType,
-		Pair:              action.Pair,
-		ID:                strconv.FormatInt(resp.OrderNumber, 10),
-		Price:             action.Price,
-		Amount:            action.Amount,
-		PostOnly:          action.PostOnly,
-		ImmediateOrCancel: action.ImmediateOrCancel,
-	}, nil
+	modResp, err := action.DeriveModifyResponse()
+	if err != nil {
+		return nil, err
+	}
+	modResp.OrderID = strconv.FormatInt(resp.OrderNumber, 10)
```

Comment:
Include fee

Reply:
I am not sure about adding this as it seems to be a fee rate in the docs [here](https://docs.poloniex.com/#moveorder). I don't have funds and actively modifying orders on Poloniex. So I don't know how that relates to the executed trades when the order has been modified and whether that needs to be handled more appropriately. 

In your other nits you suggest adding additional fields as well for this PR.  I didn't want to add extra fields but reduce fields that weren't being used in the code base. Maybe we can defer this for now until I can actively use modify like `Btc Markets` because of nuanced differences between exchanges?

As for not implicitly trusting the price/amounts data that is good point but I wanted to keep the state the same as before. Like bitmex for example we were using the modify field price and the params.OrderQty which was generated from the modify struct to make the return struct anyway. Maybe add a trello item for this?

Reply:
Yeah, given my spiel for the `order.Detail`, I'm okay with the removal of fields here - though I think fee is important and will likely be added back. The separation of `order.Modify` and having a `order.ModifyReposonse` is good too. I've been inconsistent with my responses in these PRs, but the two are on a fine line of containing information that could be misleading which is what I'm trying to avoid - though its a lot less with this Modify PR versus the Detail one. 

The main thing though is that we send `order.ModifyResponse` to the order manager and update our internal order records. By using the request information to update internally and not the real server-side data, we're updating our internal records of an order with information that may not be accurate. This was why I focused on fee information - if you change the price and amount of an order, the fee that you'd pay on the order would be changed too. If it were just a response struct that you'd hold and then call `GetOrderInfo` afterwards and update with that, then I wouldn't mind so much.

BUT, you're right, this PR is mostly keeping the functionality the same, so a Trello ticket will suffice.

Reply:
Thank you sir! 💐